app/src/main/java/com/example/music_app/data/LoginStore.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.music_app.data
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+
+private const val KEY = "access_token"
+private const val NAME = "AccessToken"
+
+
+class LoginStore(private val context: Context) {
+    companion object {
+        private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(NAME)
+        private val ACCESS_TOKEN_KEY: Preferences.Key<String> = stringPreferencesKey(KEY)
+    }
+
+    val getAccessToken: Flow<String> = context.dataStore.data.map { preferences ->
+        preferences[ACCESS_TOKEN_KEY] ?: ""
+    }
+
+    suspend fun saveToken(token: String) {
+        context.dataStore.edit { preferences ->
+            preferences[ACCESS_TOKEN_KEY] = token
+        }
+    }
+}
```

Comment:
I would recommend creating some kind of encapsulation over the DataStore by creating the class that operates methods like `readString`/`writeString` etc. Also, please make sure that this class would be singleton